index.js

```diff
@@ -149,6 +150,90 @@ class GithubScm extends Scm {
     }
 
     /**
+     * Adds the Screwdriver webhook to the Github repository
+     * @method _addWebhook
+     * @param  {Object}    config            Config object
+     * @param  {String}    config.scmUri     The SCM URI to add the webhook to
+     * @param  {String}    config.token      Service token to authenticate with Github
+     * @param  {String}    config.webhookUrl The URL to use for the webhook notifications
+     * @return {Promise}                     Resolve means operation completed without failure.
+     */
+    _addWebhook(config) {
+        const searchForHook = (scmInfo, url, page) => {
+            const getHooks = this.breaker.runCommand({
+                action: 'getHooks',
+                token: config.token,
+                params: {
+                    owner: scmInfo.owner,
+                    repo: scmInfo.repo,
+                    page,
+                    per_page: WEBHOOK_PAGE_SIZE
+                }
+            });
+            const filterForHook = getHooks.then((hooks) => {
+                let found = null;
+
+                hooks.some((hook) => {
+                    const currentHook = hoek.reach(hook, 'config.url');
+
+                    if (currentHook === url) {
+                        found = hook;
+                    }
+
+                    return found;
+                });
+
+                return found;
+            });
+
+            return Promise.all([
+                getHooks,
+                filterForHook
```

Comment:
`Promise.all` is used when you have a set of promises and you want to wait for all the promises to resolve before proceeding.

The `github.repos.getHooks` function is only called once. `getHooks` is an instance of the promise resulting from `github.repos.getHooks`. Regardless of how many instances of a promise you have, it only resolves one time, regardless of how many chains are built from it. [Promise/A+ spec](https://promisesaplus.com/)

I provided a gist in Min's comment that shows it working in a different context: https://gist.github.com/FenrirUnbound/8f18406c4e4c081607bb3fa108f3200f